pkg/util/aad/aad.go

```diff
@@ -29,33 +56,39 @@ func GetToken(ctx context.Context, log *logrus.Entry, clientID, clientSecret, te
 		AADEndpoint:  aadEndpoint,
 	}
 
-	sp, err := conf.ServicePrincipalToken()
+	spToken, err := conf.ServicePrincipalToken()
 	if err != nil {
 		return nil, err
 	}
 
-	authorizer := refreshable.NewAuthorizer(sp)
+	return spToken, nil
+}
 
+// AuthenticateServicePrincipalToken authenticates in the customer's tenant as the cluster service principal and returns a token.
+func AuthenticateServicePrincipalToken(ctx context.Context, log *logrus.Entry, authorizer refreshable.Authorizer, timeout *time.Duration) error {
 	// during credentials rotation this can take time to propagate
-	timeoutCtx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+	// it is overridable so we can have unit tests pass/fail quicker
+	ctxTimeout := DEFAULT_TIMEOUT
+	if timeout != nil {
+		ctxTimeout = *timeout
+	}
+	timeoutCtx, cancel := context.WithTimeout(ctx, ctxTimeout)
```

Comment:
@facchettos :
this cannot be done because our default timeout value is set to 5 minutes, when it comes to the test case the function will wait there till the default time which will cause to test case failure.